File: Flip Bits[15-August-2023]/solution.cpp
//  >> CPP CODE <<

class Solution{
    public:
    int maxOnes(int a[], int n)
    {
        int mx=0;
        int cnt1=0;
        
        int sum=0;
        
        for(int i=0;i<n;i++)
        {
            if(a[i]==0) sum+=1;
            else{
             cnt1++;   
             sum+=(-1);
            }
            
            mx=max(mx,sum);
            
            if(sum<0) sum=0;
        }
        
        return mx+cnt1;
        
        
    }
};

File: Flip Bits[15-August-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
    def maxOnes(self, a, n):
        sum,mx=0,0
        cnt1=0
        
        for i in range(n):
            sum+=1 if a[i]==0 else (-1)
            
            if(a[i]==1): cnt1+=1
            
            mx=max(mx,sum)
            
            if(sum<0): sum=0
            
        return cnt1+mx    